control-plane/agents/src/bin/core/controller/scheduling/mod.rs

```diff
@@ -92,6 +92,13 @@ impl NodeFilters {
             true
         }
     }
+    /// Should only attempt to use node where there are no targets for the current volume.
+    pub(crate) fn no_targets(request: &GetSuitableNodesContext, item: &NodeItem) -> bool {
+        let volume_targets = request.registry().specs().get_volume_nexuses(&request.uuid);
+        !volume_targets
+            .into_iter()
+            .any(|n| &n.lock().node == item.node_wrapper().id())
```

Comment:
1. Why is locking needed here? Even if it needed, should it be on the 'volume_targets' instead of per item?
2. Is there a possibility that there is only one healthy node at the moment and a previous nexus for the same volume on the same node is in a shutdown/deletion state?

Reply:
Because the object in case is protected by a Mutex, eg: Mutex<Obj>, so to access its inner value (Obj) you have to take the lock which gives you a guard over it,eg: Guard<Obj>. When the guard is dropped, the mutex is unlocked.

Reply:
I see, thanks for the explanation !!